sympy/physics/mechanics/tests/test_body.py

```diff
@@ -287,3 +289,30 @@ def test_apply_loads_on_multi_degree_freedom_holonomic_system():
     assert P.loads == [(P.masscenter, P.mass*g*W.y), (P.frame, (T + kT*q2)*W.z)]
     assert b.loads == [(b.masscenter, b.mass*g*W.y), (b.frame, -kT*q2*W.z)]
     assert W.loads == [(W.masscenter, (c*q1.diff() + k*q1)*W.x)]
+
+
+def test_parallel_axis():
+    N = ReferenceFrame('N')
+    m, Ix, Iy, Iz, a, b = symbols('m, I_x, I_y, I_z, a, b')
+    Io = inertia(N, Ix, Iy, Iz)
+    # Test RigidBody
+    o = Point('o')
+    p = o.locatenew('p', a * N.x + b * N.y)
+    R = Body('R', masscenter=o, frame=N, mass=m, central_inertia=Io)
+    Ip = R.parallel_axis(p)
+    Ip_expected = inertia(N, Ix + m * b**2, Iy + m * a**2,
+                          Iz + m * (a**2 + b**2), ixy=-m * a * b)
+    assert Ip == Ip_expected
+    A = ReferenceFrame('A')
+    A.orient_axis(N, N.z, 1)
+    assert trigsimp(R.parallel_axis(p, A).to_matrix(A) -
+            Ip_expected.to_matrix(A)) == zeros(3, 3)
```

Comment:
It would be helpful if you make a comment before line 306 that says in a sentence what you are testing. It's hard to read the few lines and immediately think of why this should be zero. If you have short explanations before each test then the reader can more quickly know what is going on. I recommend that for all your code additions.

Reply:
Was taking another look at it ad it should be allowed to not convert it to matrices I would say, but:
```py
from sympy import trigsimp
from sympy.physics.mechanics import inertia, Point, ReferenceFrame, RigidBody, Dyadic
N = ReferenceFrame('N')
m, Ix, Iy, Iz, a, b = symbols('m, I_x, I_y, I_z, a, b')
Io = inertia(N, Ix, Iy, Iz)
o = Point('o')
p = o.locatenew('p', a * N.x + b * N.y)
R = RigidBody('R', o, N, m, (Io, o))
Ip = R.parallel_axis(p)
Ip_expected = inertia(N, Ix + m * b**2, Iy + m * a**2,
                      Iz + m * (a**2 + b**2), ixy=-m * a * b)
assert Ip == Ip_expected
# Reference frame from which the parallel axis is viewed should not matter
A = ReferenceFrame('A')
A.orient_axis(N, N.z, 1)
assert trigsimp(R.parallel_axis(p, A) - Ip_expected) == Dyadic(0)
```
results in `AttributeError: 'function' object has no attribute 'x'`

`assert (R.parallel_axis(p, A) - Ip_expected).express(N).simplify() == Dyadic(0)` does work, since it simplifies the components of the dyadic. What shall we use the matrix conversion or `express(N).simplify()` option?

Reply:
`assert _simplify_matrix((R.parallel_axis(p, A) - Ip_expected).to_matrix(A)) == zeros(3, 3)` seems best to me. However I do find the `AttributeError` rather strange.

Reply:
The attribute error is because our physics vectors are not subclassed from SymPy's basic (and are mutable). There is a common issue that sympy functions don't work with Vectors/Dyadics.

Reply:
Ah okay, thanks for the clarification.